### ISSUE TITLE
Memory Leak in aireplay-ng

### ISSUE THREAD

**Issue by jbenden** [jbenden/aircrack-ng-migration#1146]
**Reported by Moshe Kaplan  me on 14 Jul 2013 23:42 UTC**

Memory is allocated with `malloc`, but it is possible for the function to exit without storing a reference to the allocated memory or `free`ing it.

On aireplay-ng.c, line 3199-3202:
```
if ( opt.ringbuffer )
	arp = (struct ARP_req*) malloc( opt.ringbuffer * sizeof( struct ARP_req ) );
else
	arp = (struct ARP_req*) malloc( sizeof( struct ARP_req ) );

```

This function will likely need a fair amount of restructuring to prevent memory from leaking, since there are many exit points.

Migrated-From: https://trac.aircrack-ng.org/ticket/1146

**Comment by jbenden** [comment]
**Attachment by Moshe Kaplan <me@moshekaplan.com> on 16 Jul 2013 04:39 UTC**

Uploaded file [1146_patch.diff](https://github.com/aircrack-ng/trac-attachments/blob/master/1146/20130716-043926_771-1146_patch.diff)

**Comment by jbenden** [comment]
**Comment by misterx on 14 Jul 2013 23:42 UTC**

Fixed in aircrack-ng/aircrack-ng@bd36dea6d5bbbd25f4db80e6ce58439de2d53590
